bigquery_etl/docs/mozdata/generate_mozdata_docs.py

```diff
@@ -0,0 +1,65 @@
+import os
+import yaml
+from pathlib import Path
+from jinja2 import Environment, FileSystemLoader
+
+VIEW_FILE = "view.sql"
+METADATA_FILE = "metadata.yaml"
+NON_USER_FACING_DATASET_SUFFIXES = (
+    "_derived",
+    "_external",
+    "_bi",
+    "_restricted",
+)
+SOURCE_URL = "https://github.com/mozilla/bigquery-etl/blob/master"
+
+
+def generate_mozdata_docs(out_dir, project_dir):
+    """Generate documentation for mozdata."""
+    project_doc_dir = Path(os.path.join(out_dir, "mozdata"))
+
+    data_sets = [
+        item
+        for item in os.listdir(project_dir)
+        if os.path.isdir(os.path.join(project_dir, item))
+        and all(name not in item for name in NON_USER_FACING_DATASET_SUFFIXES)
+    ]
```

Comment:
Using Path methods can be more concise:

```suggestion
    data_sets = [
        item
        for item in Path(project_dir).glob("**/*")
        if item.is_dir()
        and all(name not in item.name for name in NON_USER_FACING_DATASET_SUFFIXES)
    ]
```

Reply:
I tried using Path here, but realized that it would not work since I want the output to be a list of strings, and if I use Path I would get a list of Path objects instead. To make use of the Path objects I'd have to then convert it to string, split by "/" to get just the data set names, which defeats the purpose of reducing complexity.

Reply:
You can use `item.name` to get the filename, `item.parent.name` to get the directory name, or `str(item)` to get the path name.

Reply:
data_sets = [
        item.parent.name
        for item in Path(project_dir).glob("**/*")
        if item.is_dir()
        and all(name not in str(item.parent.name) for name in NON_USER_FACING_DATASET_SUFFIXES)
    ]

Thanks for the tips! I'm now running my script with the above code but it's taking much longer to run the scripts (5 minutes vs 10 seconds). Do you know if it's normal to downgrade in performance with `Path.glob` compared to `os.listdir` when looping like this?

Reply:
No, there shouldn't be a significant difference. Is the result correct? `item.is_dir()` means that you probably want `item.name` as the result instead of `item.parent.name`. Another explaination I can think of is that `project_dir` is set to the root, and you're traversing all the files in a virtualenv (which os.walk would be doing too, hmm).

Reply:
Yes the result is correct. `item.parent.name` because I only want the derived dataset names, not all the view names. Is it okay to use my approach instead since it's much quicker? I know it's not not as concise, would it help if I provide more comments?

Reply:
Yes, it's a nitpicky thing (mostly because I think the Path interface is generally nicer to use). Feel free to use what works; there's no definitive way of doing a thing. I am very suspicious of the difference in run time however.